### PR TITLE
Add storage endpoint to AzureEnvironment

### PR DESCRIPTION
**Pull request by freva**
# Description

I would like to add support for automatically adding scope for requests to Storage API (#39561). The main change is in `azure-resource-manager`, but in order to make that change, I guess I have to make for a new release of `azure-core-management` with this change(?).

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.** (No significant changes)
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by freva** [comment]
FYI: @weidongxu-microsoft

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java [hunk 1]
```diff
@@ -323,7 +332,9 @@ public enum Endpoint {
         /** Microsoft Graph APIs endpoint. */
         MICROSOFT_GRAPH("microsoftGraphResourceId"),
         /** Managed HSM DNS suffix. */
-        MANAGED_HSM("managedHsmDnsSuffix");
+        MANAGED_HSM("managedHsmDnsSuffix"),
+        /** Storage DNS suffix. */
+        STORAGE("storageEndpointSuffix");
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java:337]: We need to provide a default value in major clouds

https://github.com/Azure/azure-sdk-for-java/blob/c26da695af442ddba0b6b3bb867d714c34ada29d/sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java#L62-L90

**Review by weidongxu-microsoft — COMMENTED** [review]
@XiaofeiCao Please take a review.

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java [hunk 2]
```diff
@@ -258,6 +258,15 @@ public String getKeyVaultDnsSuffix() {
         return endpoints.get("keyVaultDnsSuffix");
     }
 
+    /**
+     * Gets the managed HSM DNS suffix.
+     *
+     * @return the managed HSM DNS suffix.
+     */
+    public String getManagedHmsDnsSuffix() {
```
Comment (XiaofeiCao) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java:266]: typo. `getManagedHsmDnsSuffix`.

**Comment by XiaofeiCao** [comment]
Thanks @freva . 

Yeah, for getting default scope support, guess we need to wait for new azure-core-management release(first week next month).

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java (hunk 2 above)
Comment (freva) [reply sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java]: Thanks, fixed.

**Pull request merged**